url-updater/src/main/java/com/devonfw/tools/ide/url/tool/pycharm/PycharmUrlUpdater.java

```diff
@@ -0,0 +1,59 @@
+package com.devonfw.tools.ide.url.tool.pycharm;
+
+import java.util.List;
+
+import com.devonfw.tools.ide.json.JsonMapping;
+import com.devonfw.tools.ide.url.tool.intellij.IntellijJsonObject;
+import com.devonfw.tools.ide.url.tool.intellij.IntellijUrlUpdater;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * {@link IntellijUrlUpdater} base class for Pycharm.
+ */
+public class PycharmUrlUpdater extends IntellijUrlUpdater {
```

Comment:
this hierarchy is a little strange.
Great that you want to reuse but you should extract an abstract `IdeaBasedUrlUpdater` to extend all the IDEA UrlUpdaters from.

Reply:
I've added a new `IdeaBasedUrlUpdater` now.

Reply:
great. Thanks. You could also move `getCpeVendor()` up to IdeaBasedUrlUpdater then. However, I will resolve already.